### DISCUSSION TITLE
parametric read var_sources

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#6252] [Help & Support]
Hello, for our current pipeline I am implementing something like this on pull-requests:
- find the pr author & its cluster
- obtain a k8s access to it via vault (var_source)
- deploy the app on its cluster

The only piece I am missing is the read with a  dynamic parameter, is it possible ?

I am referring to:

```json
params: {
  "GCP_JSON": "((gcp:<var>))"
}
```

where gcp is already a defined vault var_source.

What I have done is do the vaut read inside the task but I would prefer if concourse did it.

**Comment by evanchaoli** [comment]
I think it's supported. Have you tried? What problem you hit?

**Comment by schmurfy** [comment]
in fact I am not even sure how I would write that, I did not even tried the following because I doubt there is any chances it would work but I have no idea what the other options are:
```json
params: {
  "GCP_JSON": "((gcp:((.:role))))"
}
```
